### DISCUSSION TITLE
Unable to integrate pyest/pipenv project with Jenkins

### DISCUSSION THREAD

**Discussion by pandu-bhojaraj-mf** [pypa/pipenv#5832] [Q&A] [answered]
Hi All, 

1. I am trying to run my pytest frameowk project from Linux/Ubuntu Jenkin machine
2. I am able to clone code from Git repo using Jenkin job into /var/lib/bin/workspace folder
3. I am passing below two commands from Build step execute shell option

![image](https://github.com/pypa/pipenv/assets/73541850/09ac994a-973b-4ad0-89aa-28c8602d47b8)

- First command `pip install pipenv` is running fine
- Second command `pipenv install requests` is NOT executing. 
- Second command `pipenv install requests` is giving pipenv: not found **error** as below

![image](https://github.com/pypa/pipenv/assets/73541850/2da86955-5cfd-4851-b0cc-8f6fafa78f39)

Please let me know how to fix this 

Basically i would like to run below commands from Jenkins and run my test case

pipenv install -r requirements.txt --dev file which contains below commands

- pip install pipenv
- pipenv install requests
- pipenv install pytest --dev
- pipenv install selenium --dev
- pipenv install pytest-html-reporter
- pipenv install vertica_python
- pipenv install psycopg2-binary
- pipenv install pytest-dependency==0.4.0

**Comment by pandu-bhojaraj-mf** [comment, marked as answer]
This is solved from below video

Make sure you have installed `sudo apt-get install python3.x-venv` library

https://www.youtube.com/watch?v=iotloCXtIzY

#!/bin/bash
echo " ### Create Pythin  environment ### "
source scl_source enable rh-python36  # # This step is NOT required if your python is already piinting to 3.*

VIRTUAL_ENV_NAME='virtual-environment'
python -m venv $VIRTUAL_ENV_NAME

echo '### Activate Virtual Environment here ####'
source $VIRTUAL_ENV_NAME/bin/activate

echo '### Install requirements ###'

pipenv install -r requirements.txt --dev
